src/msgpack/Converters/MapConverter.cs

```diff
@@ -54,7 +54,7 @@ public override TMap Read(IMsgPackReader reader, MsgPackContext context, Func<TM
         private bool TryGetLengthFromFixMap(DataTypes type, out uint length)
         {
             length = type - DataTypes.FixMap;
-            return (type & DataTypes.FixMap) == DataTypes.FixMap;
```

Comment:
Checking for Null, Map16 and Map32 moved ahead FixMap reading (the same made for Array and String converters).